compiler/rustc_middle/src/thir/abstract_const.rs

```diff
@@ -22,7 +22,7 @@ pub enum CastKind {
 /// A node of an `AbstractConst`.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, HashStable, TyEncodable, TyDecodable)]
 pub enum Node<'tcx> {
-    Leaf(ty::Const<'tcx>),
+    Leaf(mir::ConstantKind<'tcx>),
```

Comment:
Leaf should keep being a `ty::Const`. When encountering a mir constant which can't be converted to a valtree, we should error.